docs/Examples/get-tx-from-chain.js

```diff
@@ -11,37 +11,74 @@ const logger = new Logger({
 // Create chain for testnet, specify chain directory
 const chain = new bcoin.Chain({
   logger: logger,
+  memory: true,
+  network: 'testnet'
+});
+
+const mempool = new bcoin.Mempool({ chain: chain });
+
+// Create a network pool of peers with a limit of 8 peers.
+const pool = new bcoin.Pool({
+  logger: logger,
+  chain: chain,
+  mempool: mempool,
+  maxPeers: 8
+});
+
+// Create a chain indexer which indexes tx by hash
+const indexer = new bcoin.TXIndexer({
+  logger: logger,
+  memory: true,
   network: 'testnet',
-  db: 'leveldb',
-  prefix: '/tmp/bcon-testnet',
-  indexTX: true,
-  indexAddress: true
+  chain: chain
 });
 
-(async () => {
+// Open the chain, pool and indexer
+(async function() {
   await logger.open();
+
+  await pool.open();
+
+  // Connect, start retrieving and relaying txs
+  await pool.connect();
+
+  // Start the blockchain sync.
+  pool.startSync();
+
   await chain.open();
 
+  await indexer.open();
+
   console.log('Current height:', chain.height);
 
-  const entry = await chain.getEntry(50000);
-  console.log('Block at 50k:', entry);
+  // Watch the action
+  chain.on('block', (block) => {
+    console.log('block: %s', block.rhash());
+  });
+
+  mempool.on('tx', (tx) => {
+    console.log('tx: %s', tx.rhash);
+  });
 
-  // eslint-disable-next-line max-len
-  const txhash = '4dd628123dcde4f2fb3a8b8a18b806721b56007e32497ebe76cde598ce1652af';
-  const txmeta = await chain.db.getMeta(bcoin.util.revHex(txhash));
-  const tx = txmeta.tx;
-  const coinview = await chain.db.getSpentView(tx);
+  pool.on('tx', (tx) => {
+    console.log('tx: %s', tx.rhash);
+  });
 
-  console.log(`Tx with hash ${txhash}:`, txmeta);
+  await new Promise(r => setTimeout(r, 300));
```

Comment:
What's the timeout for? Maybe add a comment to document?

Reply:
Oh, it's just to allow indexers to catch up.